core/embed/rust/src/ui/component/label.rs

```diff
@@ -63,22 +63,25 @@ where
 
     pub fn max_size(&self) -> Offset {
         let font = self.font();
-        Offset::new(font.text_width(self.text.as_ref()), font.text_max_height())
+        Offset::new(
+            font.text_width(self.text.map(|c| c)),
+            font.text_max_height(),
+        )
     }
 
     pub fn text_height(&self, width: i16) -> i16 {
         let bounds = Rect::from_top_left_and_size(Point::zero(), Offset::new(width, i16::MAX));
         self.layout
             .with_bounds(bounds)
-            .fit_text(self.text.as_ref())
+            .fit_text(self.text.map(|c| c))
             .height()
     }
 
     pub fn text_area(&self) -> Rect {
         // XXX only works on single-line labels
         assert!(self.layout.bounds.height() <= self.font().text_max_height());
         let available_width = self.layout.bounds.width();
-        let width = self.font().text_width(self.text.as_ref());
+        let width = self.font().text_width(self.text.map(|c| c));
```

Comment:
specifically, you can't go around `s/.as_ref()/.map(|c| c)/` because that's incorrect code.
what you need here is:
```suggestion
        let width = self.text.map(|c| self.font().text_width(c)));
```

Reply:
https://github.com/trezor/trezor-firmware/commit/a045dce2ae1f54bd6d8eb5fccee051fa194de545